Author: Pino Toscano <toscano.pino@tiscali.it>
Description: Disable __libc_enable_secure usage on Hurd
 __libc_enable_secure is not set as 1 when executing something as sgid
 and the differet gid is still part of the user's own groups. See also:
 - http://lists.gnu.org/archive/html/bug-hurd/2015-06/msg00051.html
 - http://lists.gnu.org/archive/html/bug-hurd/2015-07/msg00000.html
 .
 The rest of the getauxval reimplementation will do its job according
 to what p11-kit expects.
Last-Update: 2018-08-11

--- a/common/compat.c
+++ b/common/compat.c
@@ -803,7 +803,7 @@ _p11_getauxval (unsigned long type)
 	assert (type == AT_SECURE);
 
 	if (!check_secure_initialized) {
-#if defined(HAVE___LIBC_ENABLE_SECURE)
+#if defined(HAVE___LIBC_ENABLE_SECURE) && !defined(__GNU__)
 		extern int __libc_enable_secure;
 		secure = __libc_enable_secure;
 
